wgpu-hal/src/dx12/shader_compilation.rs

```diff
@@ -84,25 +84,30 @@ mod dxc {
     use std::path::PathBuf;
 
     pub(crate) struct DxcContainer {
-        pub compiler: hassle_rs::DxcCompiler,
-        pub library: hassle_rs::DxcLibrary,
+        compiler: hassle_rs::DxcCompiler,
+        library: hassle_rs::DxcLibrary,
         // Has to be held onto for the lifetime of the device otherwise shaders will fail to compile
         _dxc: hassle_rs::Dxc,
```

Comment:
I've never even considered the fact that destructor ordering for Rust structs was a thing.

This should be fine because compiler and library are created from _dxc, and validator is created from _dxil (and afaik doesn't rely on _dxc at all).

Reply:
Could you add a comment to that affect, just so it's visible to future readers that it is safe

Reply:
I rearranged it so that `_dxil` and `_dxc` are on the bottom and added a comment re: destructoring.